FlowCrypt/Models/Common/Keypair.swift

```diff
@@ -34,4 +34,39 @@ extension Keypair {
         self.allLongids = object.allLongids.map { $0 }
         self.lastModified = object.lastModified
     }
+
+    init (_ k: KeyDetails, passPhrase: String?, source: String) throws {
+        // todo - this is duplicate code from KeypairRealmObject
+        guard let privateKey = k.private, let isFullyEncrypted = k.isFullyEncrypted else {
+            throw KeypairError.missingPrivateKey("storing pubkey as private")
+        }
+        guard isFullyEncrypted else {
+            throw KeypairError.notEncrypted("Will not store Private Key that is not fully encrypted")
+        }
+        guard k.ids.isNotEmpty else {
+            throw KeypairError.missingKeyIds
+        }
+        // end duplicate
+        self.primaryFingerprint = k.primaryFingerprint
+        self.private = privateKey
+        self.public = k.public
+        self.passphrase = passPhrase
+        self.source = source
+        self.allFingerprints = k.fingerprints
+        self.allLongids = k.ids.map { $0.longid }
+        guard let lastModified = k.lastModified else {
+            // todo - make a new error like `keyMissingSelfSignature`
+            throw KeyMethodsError.parsingError
+        }
```

Comment:
This new init method could probably be used wherever we have generated a new key during setup and are storing it. There is probably similar code written in the VC, or somewhere in setup procedure in "generate key" flow.

Reply:
#1589